api/utils/mwasHypothesisGenerator.mjs

```diff
@@ -0,0 +1,117 @@
+import { getMWASResults } from "./mwas.mjs"
+import { runCypherQuery } from '../clients/neo4j.mjs';
+import { runLLMCompletion } from '../clients/oai.mjs';
+
+export const getMwasHypothesis = async (bioprojects, targetVirusFamilies, pageStart, pageEnd, identifiers) => {
+
+    let model, role;
+    model = 'gpt4o';
+    if (model === 'gpt4o') {
+        role = 'system';
+    } else {
+        role = 'assistant'; // o1
+    }
+
+
+    const results = await getMWASResults(bioprojects, targetVirusFamilies, pageStart, pageEnd);
+    const TARGET_QUERY = "Eimeria"
+    const test = {text: "Display this"}
+
+    const bioprojectRecords = await runCypherQuery(
+        `MATCH (n:BioProject)
+        WHERE n.bioProject IN [${bioprojects.map((id) => `'${id}'`).join(', ')}]
+        RETURN n
+        ORDER BY size(n.description) DESC
+        LIMIT 20
+        `,
+    );
+
+    const serializeBioProjectRecord = (record) => {
+        if (!record?.n) {
+            return {};
+        }
+        const { bioProject: id, name, title, description } = record.n.properties;
+        return JSON.stringify({ id, name, title, description });
+    };
+    const maxTotalLength = 10000;
+    const serializedBioprojects = [];
+    let totalLength = 0;
+    for (const serialized of bioprojectRecords.map(serializeBioProjectRecord)) {
+        if (totalLength + serialized.length > maxTotalLength) {
+            break;
+        }
+        serializedBioprojects.push(serialized);
+        totalLength += serialized.length;
+    }
+
+    // contextual_info = bioproject_df[bioproject_df['bioproject'] != result['bioproject_id']]
+    // contextual_info = contextual_info.to_dict(orient='records')
```

Comment:
try to clean up comments/old code

Reply:
Removed, see latest commit: d56d4bbb08682c5672d2ba3d502e6597fc476ab6